bin/obsidian_server.py

```
@@ -77,6 +77,23 @@ def do_POST(self):
+        home_base = os.path.realpath(os.environ.get('HOME', '/tmp'))
+        cwd = home_base
+        try:
+            # Validate raw_cwd relative to home_base
+            norm_raw = os.path.normpath(str(raw_cwd))
+            if os.path.isabs(norm_raw):
+                target_path = os.path.realpath(norm_raw)
+            else:
+                target_path = os.path.realpath(os.path.join(home_base, norm_raw))
+
+            # CodeQL sanitizer guard: check realpath boundary alignment
+            if target_path.startswith(home_base) and os.path.commonpath([home_base, target_path]) == home_base and os.path.isdir(target_path):
```

Pylint (4.0.7)

[convention] Line too long (142/100)

(C0301)